File: elixir/lasagna/lib/lasagna.ex
defmodule Lasagna do
  # Please define the 'expected_minutes_in_oven/0' function
  @spec expected_minutes_in_oven() :: number()
  def expected_minutes_in_oven, do: 40

  # Please define the 'remaining_minutes_in_oven/1' function
  @spec remaining_minutes_in_oven(number()) :: number()
  def remaining_minutes_in_oven(minutes), do: expected_minutes_in_oven() - minutes

  # Please define the 'preparation_time_in_minutes/1' function
  @spec preparation_time_in_minutes(number()) :: number()
  def preparation_time_in_minutes(layers), do: layers * 2

  @spec total_time_in_minutes(number(), number()) :: number()
  # Please define the 'total_time_in_minutes/2' function
  def total_time_in_minutes(layers, minutes), do: preparation_time_in_minutes(layers) + minutes

  @spec alarm() :: String.t()
  # Please define the 'alarm/0' function
  def alarm, do: "Ding!"
end

File: php/lasagna/Lasagna.php
<?php

declare(strict_types=1);

class Lasagna
{
    // Please define the 'expectedCookTime()' function
    function expectedCookTime() 
    {
        return 40;
    }

    // Please define the 'remainingCookTime($elapsed_minutes)' function
    function remainingCookTime(int $minutes_in_oven)
    {
        return Lasagna::expectedCookTime() - $minutes_in_oven;
    }

    // Please define the 'totalPreparationTime($layers_to_prep)' function
    function totalPreparationTime(int $layers)
    {
        return $layers * 2;
    }

    // Please define the 'totalElapsedTime($layers_to_prep, $elapsed_minutes)' function
    function totalElapsedTime(int $layers, int $minutes_in_oven)
    {
        return Lasagna::totalPreparationTime($layers) + $minutes_in_oven;
    }

    // Please define the 'alarm()' function
    function alarm() {
        return "Ding!";
    }
}
